src/tests/Interop/Swift/SwiftErrorHandling/SwiftErrorHandling.cs

```diff
@@ -64,6 +64,7 @@ private static string GetErrorMessageFromSwift(SwiftError error, int length)
         IntPtr pointer = GetErrorMessage(error.Value);
         byte[] byteArray = new byte[length];
         Marshal.Copy(pointer, byteArray, 0, length);
+        Marshal.FreeCoTaskMem(pointer);
```

Comment:
This is surprising to me. I would expect `NativeLibrary.Free()` instead. I don't think we want to be reusing `Marshal.FreeCoTaskMem()` in Swift interop scenarios. It might also be worth adding to our design documentation the memory manager we are expecting to use. Is it simply C's `malloc` / `free`? Do we need a new API for Swift runtime memory rules?

Reply:
Oh, I've just realized that the `FreeCoTaskMem` is for COM interop.

Reply:
Please take a look at https://github.com/dotnet/designs/pull/307.